src/ivoc/ivocvect.cpp

```diff
@@ -3875,198 +3876,9 @@ void Vector_reg() {
-                    }
-                } while (left_ptr <= right_ptr);
-            }
-
-            /* Set up pointers for next iteration.  First determine whether
-               left and right partitions are below the threshold size. If so,
-               ignore one or both.  Otherwise, push the larger partition's
-               bounds on the stack and continue sorting the smaller one. */
-
-            if ((right_ptr - lo) <= max_thresh) {
-                if ((hi - left_ptr) <= max_thresh) /* Ignore both small partitions. */
-                    POP(lo, hi);
-                else /* Ignore small left partition. */
-                    lo = left_ptr;
-            } else if ((hi - left_ptr) <= max_thresh) /* Ignore small right partition. */
-                hi = right_ptr;
-            else if ((right_ptr - lo) > (hi - left_ptr)) /* Push larger left partition indices. */
-            {
-                PUSH(lo, right_ptr);
-                lo = left_ptr;
-            } else /* Push larger right partition indices. */
-            {
-                PUSH(left_ptr, hi);
-                hi = right_ptr;
-            }
-        }
-    }
-
-    /* Once the BASE_PTR array is partially sorted by quicksort the rest
-       is completely sorted using insertion sort, since this is efficient
-       for partitions below MAX_THRESH size. BASE_PTR points to the beginning
-       of the array to sort, and END_PTR points at the very last element in
-       the array (*not* one beyond it!). */
-
-
-    {
-        int* end_ptr = base_ptr + 1 * (total_elems - 1);
-        int* run_ptr;
-        int* tmp_ptr = base_ptr;
-        int* thresh = (end_ptr < (base_ptr + max_thresh)) ? end_ptr : (base_ptr + max_thresh);
-
-        /* Find smallest element in first threshold and place it at the
-           array's beginning.  This is the smallest array element,
-           and the operation speeds up insertion sort's inner loop. */
-
-        for (run_ptr = tmp_ptr + 1; run_ptr <= thresh; run_ptr += 1)
-            if (cmp(vec[*run_ptr], vec[*tmp_ptr]) < 0)
-                tmp_ptr = run_ptr;
-
-        if (tmp_ptr != base_ptr)
-            SWAP(tmp_ptr, base_ptr);
-
-        /* Insertion sort, running from left-hand-side up to `right-hand-side.'
-           Pretty much straight out of the original GNU qsort routine. */
-
-        for (run_ptr = base_ptr + 1; (tmp_ptr = run_ptr += 1) <= end_ptr;) {
-            while (cmp(vec[*run_ptr], vec[*(tmp_ptr -= 1)]) < 0)
-                ;
-
-            if ((tmp_ptr += 1) != run_ptr) {
-                int* trav;
-
-                for (trav = run_ptr + 1; --trav >= run_ptr;) {
-                    int c = *trav;
-                    int *hi, *lo;
-
-                    for (hi = lo = trav; (lo -= 1) >= tmp_ptr; hi = lo)
-                        *hi = *lo;
-                    *hi = c;
-                }
-            }
-        }
-    }
+    std::sort(base_ptr, base_ptr + total_elems, [&](const int& a, const int& b) {
```

Comment:
```suggestion
    std::sort(base_ptr, base_ptr + total_elems, [&](int a, int b) {
```

Reply:
I followed the signature of `comp` here: https://en.cppreference.com/w/cpp/algorithm/sort

Reply:
Since a move is equivalent to a copy for `int` we can just follow usual conventions and pass an int by value.

Reply:
done